community/cypher/frontend-3.1/src/main/scala/org/neo4j/cypher/internal/frontend/v3_1/ast/CaseExpression.scala

```diff
@@ -37,3 +39,9 @@ case class CaseExpression(expression: Option[Expression], alternatives: Seq[(Exp
     } chain this.specifyType(possibleTypes)
   }
 }
+
+object CaseExpression {
+  def apply(expression: Option[Expression], alternatives: List[(Expression, Expression)], default: Option[Expression])(position: InputPosition):CaseExpression =
```

Comment:
Isn't it preferable to take an `IndexedSeq` as parameter type instead of `List`, and make callers convert? Or did you do it this way to not change an abundance of callers? I'm thinking there's some calling code that obliviously creates a `List` and sends it here, only to have it converted to `IndexedSeq`, where creating the latter immediately would've been better (it may be a micro-optimisation).